File: Python/Problems/Problems/problem2109.py
from typing import List

from .problem import Problem

class Problem2109(Problem):
    def test(self) -> bool:
        s = "LeetcodeHelpsMeLearn"
        spaces = [8, 13, 15]

        expected = "Leetcode Helps Me Learn"

        result = self.addSpaces(s, spaces)

        print(result)

        return result == expected

    def addSpaces(self, s: str, spaces: List[int]) -> str:
        index = 0
        result = []
        for space in spaces:
            result.append(s[index:space])
            result.append(" ")
            index = space

        result.append(s[index:])
        return ''.join(result)

File: C++/Problems/Problems/problem_2109.cpp
#include "problem_2109.h"

#include <sstream>

bool problem_2109::test() {
    std::string s = "LeetcodeHelpsMeLearn";
    std::vector spaces { 8, 13, 15 };

    std::string expected = "Leetcode Helps Me Learn";

    auto result = addSpaces(s, spaces);

    std::cout << result << std::endl;

    return result == expected;
}

std::string problem_2109::addSpaces(std::string s, std::vector<int> &spaces) {
    int index = 0;
    std::ostringstream result;
    for (auto space : spaces)
    {
        result << s.substr(index, space - index);
        result << " ";
        index = space;
    }

    result << s.substr(index);
    return result.str();
}
